app/features/quizzify/tools.py

```diff
@@ -157,22 +402,44 @@ def load(self, tool_files: List[ToolFile]) -> List[Document]:
                 response = requests.get(url)
                 parsed_url = urlparse(url)
                 path = parsed_url.path
+                self.expected_file_type = tool_file.file_type
+                section_start = tool_file.section_start
+                section_end = tool_file.section_end or float('inf')
+                specific_list = tool_file.specific_list
+                if(type(section_start) == float):
+                    if(section_start >= section_end):
+                        raise LoaderError(f"section_start must be less than section_end")
+                    if(section_start < 0):
+                        section_start = 0
+                
 
                 if response.status_code == 200:
-                    # Read file
-                    file_content = BytesIO(response.content)
-
-                    # Check file type
-                    file_type = path.split(".")[-1]
-                    if file_type != self.expected_file_type:
-                        raise LoaderError(f"Expected file type: {self.expected_file_type}, but got: {file_type}")
+                    if(self.expected_file_type not in ["youtube","web_url"]):
+                        # Read file
+                        file_content = BytesIO(response.content)
+
+                        # Check file type
+                        file_type = path.split(".")[-1]
+                        if file_type != self.expected_file_type:
+                            raise LoaderError(f"Expected file type: {self.expected_file_type}, but got: {file_type}")
+
+                        # Append to Queue
+                        queued_files.append((file_content, file_type,specific_list,section_start,section_end))
+                    else:
+                        if("youtube" in url):
+                            if(self.expected_file_type != "youtube"):
```

Comment:
If not done already, make sure can handle youtube transcript API edge cases like what if there are no youtube transcripts available or none in english

Reply:
I added handling for these edge cases in the YouTubeLoader